datasource/src/test/scala/quasar/physical/mongo/MongoScalarStagesInterpreterSpec.scala

```diff
@@ -66,12 +66,29 @@ class MongoScalarStagesInterpreterSpec
     }
   }
 
+  "Pivot special" >> {
+    "no unnecessary undefined pairs" in {
+      val input = ldjson("""
+        { "a": 1 }
+        12
+        { "b": 2 }
+        """)
+      val expected = ldjson("""
+        ["a", 1]
+        ["b", 2]
+        """)
+      val targets = Pivot(IdStatus.IncludeId, ColumnType.Object)
+      evalPivot(targets, input) must bestSemanticEqual(expected)
+    }
+  }
+
   "Cartesian special" >> {
     "cross fields when some are undefined" in {
```

Comment:
I'll move this one over to `ScalarStagesSpec` as well. We don't need to hold up merge of this PR though.

Reply:
This is very specific test actually, during development I got `["whatever_pivot_undefined", "whatever_pivot_undefined"]`, I don't think that this is common case for all interpreters

Reply:
If there is a mongo-specific reason to test this here, then :+1: Maybe add a comment here about those undefined values so we know why this test is important.

Reply:
I was thinking more actually, and I'd like to have this test in `ScalarStageSpec`. Even though mongo hit this case for reasons specific to the mongo implementation, it's still a valid test for all interpreters. And the more tests that we can have over in `ScalarStageSpec` the better, because you never know what corner cases different impls will have.

Reply:
Added here https://github.com/slamdata/quasar/pull/4101/files#diff-018b1055badf3c3df25b2b218bb1abcbR2162